### PR TITLE
Multi auth keys part 2

### PR DESCRIPTION
**Pull request by aliuk2012**
#### What problem does the pull request solve?

- adds an optional description to the token when it is created
- adds the tokens owner to the logs if the request has used one of the tokens
- adds an endpoint that will deactivate a specific token 
- renames token column to token_digest

Trello card:

#### Checklist

- [ ] I've used the pull request template
- [ ] I've linked this PR to the relevant issue (if mission work)
- [ ] I've written unit tests for these changes (if code change)
- [ ] I've updated the documentation in (If any documentation requires updating)
  - [ ] README.md
  - [ ] Elsewhere (please link)

### CODE REVIEW

**Comment by lfdebrux** [comment]
I think we should consider adding some kind of identifier to the front of our tokens so that it is easy to identify from the token that it is for our api. Something like how GitHub personal access tokens start with `ghp_`, or how Notify puts the name of the api key at the start. This will make it easier to catch tokens with something detect-secrets.

File: app/controllers/api/v1/access_tokens_controller.rb [hunk 1]
```diff
@@ -14,9 +19,23 @@ def create
     end
   end
 
+  def deactivate
+    @access_token = AccessToken.find(token_deactivate_params)
+    @access_token.update!(deactivated_at: Time.zone.now)
+    render json: { status: "`#{@access_token.owner}` has been deactivated" }.to_json, status: :ok
```
[the remark this answers is not in the document]
Comment (aliuk2012) [reply app/controllers/api/v1/access_tokens_controller.rb:25]: @DavidBiddle but what if the one person left is a token that hadn't been disabled and the dev has already left? 

Either way, if we got into that situation we could create a workaround to get us back into the service

[the remark this answers is not in the document]
Comment (DavidBiddle) [reply app/controllers/api/v1/access_tokens_controller.rb:25]: Yeah, I guess we don't necessarily need a tech solution for this - maybe what we really need are some reasonably robust processes around this!

[the remark this answers is not in the document]
Comment (SamJamCul) [reply app/controllers/api/v1/access_tokens_controller.rb:25]: Yeah happy to shift the burden out of the code, as it wouldn't be that clear why it's in here

**Review by SamJamCul — APPROVED** [review]

**Comment by aliuk2012** [comment]
> I think we should consider adding some kind of identifier to the front of our tokens so that it is easy to identify from the token that it is for our api. Something like how GitHub personal access tokens start with `ghp_`, or how Notify puts the name of the api key at the start. This will make it easier to catch tokens with something like [detect-secrets](https://trello.com/c/YcSSS8kB).

I like your suggestion about prefixing our tokens. I think we could retro-fit that feature to our tokens at another stage.
We can add it for future work on this feature. We will have to iterate on it  i.e removing the old way of authentication and tidying up the headers that are used

**Pull request merged**